docs/decisions/0005-architecture-and-data-modeling.rst

```diff
@@ -0,0 +1,149 @@
+  2. **REST API for Authorization Checks**: A RESTful interface that allows services to perform authorization checks over HTTP.
+  3. **Policy Management Interface**: Tools or interfaces for administrators to manage authorization policies.
+  4. **Caching and Performance Optimization**: Mechanisms to optimize performance for frequent authorization checks: caching strategies per decision / per policy.
+  5. **Deployment and Configuration**: Support for configuring the authorization engine and policies in different environments.
+- **Authorization Clients**: Services or components that request authorization decisions via the enforcement API.
+
+Here is an overview diagram of the architecture:
+
+.. image:: ../_images/architecture-overview.png
+   :alt: Architecture Overview
+   :align: center
+   :width: 600px
+
+This architecture is supported by detailed design decisions outlined below.
+
+Decision
+********
+
+#. Framework Components and Responsibilities
+============================================
+
+Casbin as the Authorization Engine for Open edX
+-----------------------------------------------
+- Use the Authorization Engine (Casbin) as the core component for evaluating authorization requests based on defined policies.
+- The Authorization Engine is responsible for loading policies, evaluating requests, and returning decisions (allow/deny).
+- The model configuration (``model.conf``) and policy will be managed centrally as the source of truth for authorization to ensure consistency.
+
+A Dedicated Open edX Layer for Authorization
+---------------------------------------------
+- Implement an Open edX-specific layer that encapsulates all authorization logic by interacting with the Authorization Engine, ensuring that services interact with a consistent interface.
+- The Open edX Layer will provide a stable Enforcement API that abstracts Casbin internals, allowing services to request authorization decisions without needing to understand Casbin specifics.
+- Implement a Policy Management API within the Open edX Layer to allow administrators to manage and update authorization policies centrally.
+- The Open edX Layer will be implemented as a shared library, a Django app or a Tutor plugin or all three, depending on the what is most suitable. All of these components will be versioned and maintained as part of the Open edX ecosystem as a whole.
+- All modifications to the Authorization Engine configuration (model, adapters, etc.) should be done through the Open edX Layer, so no forks of Casbin are needed.
+
+Interact with the Policy Store via the Open edX Layer
+------------------------------------------------------
+- The policy store, which can be a database or configuration files (e.g., ``authz.policy``) will be accessed and managed through the Open edX Layer. No direct access to the policy store should be made by services.
```

Comment:
I think we should be more explicit when talking about the policy store that it is database that is populated via config file for immutable system wide permissions and also managed via API and UI for dynamic permissions, role configuration, and role assigment... at least if I'm understanding things correctly. 😄